src/drone.rs

```diff
@@ -0,0 +1,340 @@
+//! The `drone` module provides an object for launching a Solana Drone,
+//! which is the custodian of any remaining tokens in a mint.
+//! The Solana Drone builds and send airdrop transactions,
+//! checking requests against a request cap for a given time time_slice
+//! and (to come) an IP rate limit.
+
+use signature::{KeyPair, PublicKey};
+use std::io;
+use std::io::{Error, ErrorKind};
+use std::net::{IpAddr, SocketAddr, UdpSocket};
+use std::time::Duration;
+use thin_client::ThinClient;
+use transaction::Transaction;
+
+pub const TIME_SLICE: u64 = 60;
+pub const REQUEST_CAP: u64 = 500;
+pub const SMALL_BATCH: i64 = 50;
+pub const TPS_BATCH: i64 = 5_000_000;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum DroneRequest {
+    GetAirdrop {
+        request_type: DroneRequestType,
```

Comment:
@CriesofCarrots, thoughts on this one?

Reply:
I liked the preset batches because I thought it would make it easier to calculate how long the reserve would last on a public testnet, but I suppose the client demo code could do that instead. If the airdrop request size is variable, do you think it still makes sense to limit by number of requests per time slice?

Reply:
I'd prefer the drone govern the number of tokens distributed per time slice. For TPS-benchmark-sized requests, we can either:

* Whitelist certain public keys
* Put those tokens in the genesis block
* Give the mint's key to humans that request it from our humans